public/js/actions/breakpoints.js

```diff
@@ -187,28 +187,31 @@ function toggleAllBreakpoints(shouldDisableBreakpoints: Boolean) {
  *        The condition to set on the breakpoint
  */
 function setBreakpointCondition(location: Location, condition: string) {
-  throw new Error("not implemented");
-
-  // return ({ dispatch, getState, client }) => {
-  //   const bp = getBreakpoint(getState(), location);
-  //   if (!bp) {
-  //     throw new Error("Breakpoint does not exist at the specified location");
-  //   }
-  //   if (bp.get("loading")) {
-  //     // TODO(jwl): when this function is called, make sure the action
-  //     // creator waits for the breakpoint to exist
-  //     throw new Error("breakpoint must be saved");
-  //   }
-
-  //   return dispatch({
-  //     type: constants.SET_BREAKPOINT_CONDITION,
-  //     breakpoint: bp,
-  //     condition: condition,
-  //     [PROMISE]: Task.spawn(function* () {
-  //       yield client.setBreakpointCondition(bp.get("id"), condition);
-  //     })
-  //   });
-  // };
+  // throw new Error("not implemented");
+
+  return ({ dispatch, getState, client }) => {
+    const bp = getBreakpoint(getState(), location);
+    if (!bp) {
+      throw new Error("Breakpoint does not exist at the specified location");
+    }
+    if (bp.loading) {
+      // TODO(jwl): when this function is called, make sure the action
+      // creator waits for the breakpoint to exist
+      throw new Error("breakpoint must be saved");
```

Comment:
There no assertions here though; it should always just throw an error. `assert` may evolve in the future to only happen in dev or somethings. But this should always through.